include/motor-control/core/brushed_motor/brushed_motor_interrupt_handler.hpp

```diff
@@ -50,21 +58,72 @@ class BrushedMotorInterruptHandler {
         return queue.has_message_isr();
     }
 
+    // upon advice from hardware, 0.01mm is a good limit for percision
+    auto acceptable_position_error() -> int32_t {
+        return int32_t(gear_conf.get_encoder_pulses_per_mm() * 0.01);
+    }
+
+    auto controlled_move_to(int32_t encoder_position) -> int32_t {
+        int32_t move_delta = hardware.get_encoder_pulses() - encoder_position;
+        uint32_t old_control_pwm = current_control_pwm;
+        // pass through early if we're already within acceptable position
+        if (std::abs(move_delta) < acceptable_position_error()) {
+            current_control_pwm = 0;
+        } else {
+            if (move_delta < 0) {
+                hardware.positive_direction();
+            } else if (move_delta > 0) {
+                hardware.negative_direction();
+            }
+            double pid_output = hardware.update_control(move_delta);
+            // The min and max PWM values for an active duty move are hardware
+            // dependent so let the driver decide the bounds of the pid output
+            // take the abs of the pid output because the pwm value has to be
+            // always positive
+            current_control_pwm = driver_hardware.pwm_active_duty_clamp(
+                int(std::abs(pid_output)));
+        }
+        if (old_control_pwm != current_control_pwm) {
+            driver_hardware.update_pwm_settings(current_control_pwm);
+        }
+        return move_delta;
+    }
+
     void run_interrupt() {
         if (!has_active_move && has_messages()) {
             update_and_start_move();
         }
         if (has_active_move) {
-            if (buffered_move.stop_condition ==
-                    MoveStopCondition::limit_switch &&
-                limit_switch_triggered()) {
-                homing_stopped();
-            } else {
-                if (is_sensing() && is_idle) {
-                    finish_current_move(
-                        AckMessageId::complete_without_condition);
-                }
+            switch (buffered_move.stop_condition) {
```

Comment:
this is getting a little complex; maybe we should split it into like an `execute_active_move()` subfunction and a `execute_idle()` subfunction?